### PR TITLE
fix: allow `RecvPacket` and `WriteAcknowledgement` when channel is `FLUSHING` or `FLUSHCOMPLETE`

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

While doing testing with hermes Luca an I found that hermes couldn't relay packets when the channel end was `FLUSHING` because `RecvPacket` RPC handler failed on `WriteAcknowledgement`.

closes: #XXXX

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by DimitrisJim — APPROVED** [review]
e2e saving the day it seems, wild how this wasn't caught earlier. I _think_ this was a defensive check to begin with, hence why spec didn't have it (or update it). 

lgtm

**Review by charleenfei — APPROVED** [review]
really good work on this find @crodriguezvega!

**Review by chatton — APPROVED** [review]
LGTM thanks @crodriguezvega

File: modules/core/04-channel/keeper/packet.go [hunk 1]
```diff
@@ -287,11 +287,8 @@ func (k Keeper) WriteAcknowledgement(
 		return errorsmod.Wrap(types.ErrChannelNotFound, packet.GetDestChannel())
 	}
 
-	if !channel.IsOpen() {
-		return errorsmod.Wrapf(
-			types.ErrInvalidChannelState,
-			"channel state is not OPEN (got %s)", channel.State.String(),
-		)
+	if !slices.Contains([]types.State{types.OPEN, types.FLUSHING}, channel.State) {
```
Comment (colin-axner) [inline modules/core/04-channel/keeper/packet.go:290]: This should contained `FLUSHCOMPLETE` as well? If we finish flushing our packet sends, the counterparty might not have finished flushing theirs (we still have packets to receive)

Comment (damiannolan) [reply modules/core/04-channel/keeper/packet.go:290]: we would also need to add `FLUSHCOMPLETE` to recvPacket then here:

https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/keeper/packet.go#L121

Comment (damiannolan) [reply modules/core/04-channel/keeper/packet.go:290]: `FLUSHCOMPLETE` says we are finished processing outbound packets sent from our side, but we must be able to support processing (recving) packets which were sent from the other channel end. 

So we should add `FLUSHCOMPLETE` to recvPacket and writeAck, but acknowledgePacket should only work in `OPEN` and `FLUSHING`.

Comment (crodriguezvega) [reply modules/core/04-channel/keeper/packet.go:290]: Yes, that's right. Good catch!

Comment (crodriguezvega) [reply modules/core/04-channel/keeper/packet.go]: Updated. I will look into having an e2e for this.

**Review by colin-axner — APPROVED** [review]
Nice, thank you! I wonder if this will help the e2e's pass

**Pull request merged**